src/Di/Inject.php

```diff
@@ -20,4 +20,12 @@
      * @var bool
      */
     public $optional = false;
```

Comment:
> The annotation parser check if the annotation constructor has arguments, if so then we will pass the value array, otherwise will try to inject values into public properties directly:

http://doctrine-common.readthedocs.org/en/latest/reference/annotations.html#inject-annotation-values

This is the spec of Doctrine annotation.